lib/sitemap.coffee

```diff
@@ -56,18 +56,19 @@ module.exports = exports = (argv) ->
         linkRe = /\[\[([^\]]+)\]\]/g
         match = undefined
         while (match = linkRe.exec(currentItem.text)) != null
-          collaborativeLinks.add asSlug match[1]
+          if not collaborativeLinks.has(asSlug(match[1]))
+            collaborativeLinks.set(asSlug(match[1]), currentItem.id)
       catch err
         console.log "METADATA *** #{wikiName} Error extracting links from #{currentIndex} of #{JSON.stringify(array)}", err.message
       collaborativeLinks
 
     try
-      pageLinksSet = page.story.reduce( extractPageLinks, new Set())
+      pageLinksMap = page.story.reduce( extractPageLinks, new Map())
     catch err
       console.log "METADATA *** #{wikiName} reduce to extract links on #{slug} failed", err.message
     #
-    if pageLinksSet.size > 0
-      pageLinks = Array.from(pageLinksSet)
+    if pageLinksMap.size > 0
+      pageLinks = Object.fromEntries(pageLinksMap)
```

Comment:
I'm not seeing the value of making this an object. It would seem that any client that fetched a referencing page could discover the reference for itself. This then would be simplifying the client by adding enduring complexity to a network schema.

Reply:
That might be a hangover from an earlier pass where a `Set` was used, which needs to be converted before inserting into JSON.

Reply:
> It would seem that any client that fetched a referencing page could discover the reference for itself. This then would be simplifying the client by adding enduring complexity to a network schema.

Not sure what you are getting at! Having a list of links on a page so we can construct the a list of links to a page to show, or including the item on the page that contains the link. Without the former the client would have to scrape every page in the neighbourhood to construct the list, or the later which would require parsing the page and searching the DOM to fixup the backlink flags on all the pages in the lineup that that newly loaded page has links to.

Reply:
Ok, I was clearly confusing in-links with out-links. The out-links on a page are noted by the slug and item id that contained it. To find in-links one must look at every sitemap but not every page in the site. The predicate, does this sitemap page refer to a given slug, is made more efficient by looking up a key in the links object as opposed to searching a list of out-links. That is the advantage?